netutils/config/parser.py

```diff
@@ -331,6 +331,100 @@ def build_config_relationship(self) -> t.List[ConfigLine]:
+            ...    network 172.17.1.0 mas'''
+            >>> bgp_conf = BaseSpaceConfigParser(str(config)).find_all_children(pattern="router bgp", match_type="regex")
+            >>> print(bgp_conf)
+            ['router bgp 45000', '  address-family ipv4 unicast', '   neighbor 192.168.1.2 activate', '   network 172.17.1.0 mas']
+        """
+        config = []
+        for cfg_line in self.build_config_relationship():
+            parents = cfg_line.parents[0] if cfg_line.parents else None
+            if (
+                parents
+                and self._match_type_check(parents, pattern, match_type)
+                or self._match_type_check(cfg_line.config_line, pattern, match_type)
+            ):
+                config.append(cfg_line.config_line)
+        return config
+
+    def find_children_w_parents(
+        self, parent_pattern: str, child_pattern: str, match_type: str = "exact"
+    ) -> t.List[str]:
+        """Returns configuration part for a specific pattern including parents and children.
+
+        Args:
+            parent_pattern: pattern that describes parent.
+            child_pattern: pattern that describes child.
+            match_type (optional): Exact or regex. Defaults to "exact".
+
+        Returns:
+            configuration under that parent pattern.
+        Example:
+            >>> config = '''
+            ... router bgp 45000
+            ...   address-family ipv4 unicast
+            ...    neighbor 192.168.1.2 activate
+            ...    network 172.17.1.0 mas'''
+            >>> bgp_conf = BaseSpaceConfigParser(str(config)).find_children_w_parents(parent_pattern="router bgp", child_pattern="  address-family", match_type="regex")
+            >>> print(bgp_conf)
+            ['  address-family ipv4 unicast', '   neighbor 192.168.1.2 activate', '   network 172.17.1.0 mas']
+        """
+        config = []
+        potential_parents = [
+            elem.parents[0]
+            for elem in self.build_config_relationship()
+            if self._match_type_check(elem.config_line, child_pattern, match_type)
+        ]
+        for cfg_line in self.build_config_relationship():
+            parents = cfg_line.parents[0] if cfg_line.parents else None
+            if parents in potential_parents and self._match_type_check(parents, parent_pattern, match_type):
+                config.append(cfg_line.config_line)
+        return config
+
+    # def find_blocks():
+    # def find_children():
+    # def find_children_w_parents():
+    # def find_interface_objects():
+    # def find_lineage():
+    # def find_lines():
+    # def find_object_branches():
+    # def find_objects():
+    # def find_objects_dna():
+    # def find_objects_w_all_children():
+    # def find_objects_w_child():
+    # def find_objects_w_missing_children():
+    # def find_objects_w_parents():
+    # def find_objects_wo_child():
+    # def find_parents_w_child():
+    # def find_parents_wo_child():
```

Comment:
Are these future WIP for this PR?  If not I'd prefer to document these in a issue versus inside the source code.

Reply:
Yes, future WIP. Will document elsewhere. Deleted.